controllers/cloudinit/scripts/00-configure-snapstore-proxy.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash -xe
+
+# Usage:
+#   $0 $snapstore-domain $snapstore-id
+#
+# Assumptions:
+#   - snapd is installed
+
+if [ "$#" -ne 2 ] || [ -z "${1}" ] || [ -z "${2}" ] ; then
+  echo "Using the default snapstore"
+  exit 0
+fi
+
+while ! snap install curl; do
+  echo "Failed to install curl, will retry"
+  sleep 5
+done
+
+while ! /snap/bin/curl -sL http://"${1}"/v2/auth/store/assertions | snap ack /dev/stdin ; do
```

Comment:
why is this required? curl should come by default.

Reply:
I do not think curl is expected to be in an capi image. In any case, the revised version of this PR first checks if curl is available and if it is not found we pull it from the snapstore.